packages/flutter_tools/lib/src/commands/attach.dart

```diff
@@ -56,7 +54,16 @@ class AttachCommand extends FlutterCommand {
       ..addOption(
         'debug-port',
         help: 'Device port where the observatory is listening.',
-      )..addOption('pid-file',
+      )..addOption(
```

Comment:
Could we also infer this from the device + applicationBundle?

Reply:
A device could have multiple applications running on it.  I'm not familiar with how that's implemented though...

Reply:
What I mean is, we already have the concept of an [ApplicationPackage](https://github.com/flutter/flutter/blob/master/packages/flutter_tools/lib/src/application_package.dart) which contains enough information (when combined with the proper Device subclass) to start the app. Presumably we could use this information gathered from the user's pubspec/project files (if present) to figure out the app id.

This doesn't necessarily help for google3, but maybe we can feed some more information through from blaze and the device information to skip it as well.

Reply:
Attach doesn't start an app though, it looks for started apps.

Reply:
but you know the device you are connecting to at this point, and you might be in the root of a directory which can infer the application package structure. That doesn't mean we can't allow a user to specify the id manually, but I don't understand why we couldn't infer which observatory to attach.